File: src/ProjectR/content/scripts/spells/MirrorImage.lua
function GetTargetType()
    return Single
end

function GetName()
    return "Mirror Image"
end

function GetDescription()
    return "Creates two clones of the target to fight by your side. Clones use the same spells as the target."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 2
end

function GetDelay()
  return 0
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
    SummonMinionCopy(d, "Mirror Image")
    SummonMinionCopy(d, "Mirror Image")
end

File: src/ProjectR.Scripting/content/scripts/spells/MirrorImage.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class MirrorImage : SpellScriptBase
    {
        public override string Name { get { return "Mirror Image"; } }
        public override string Description { get { return "Creates two clones of the target to fight by your side. Clones use the same spells as the target."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 200; } }
        public override double Delay { get { return 0; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            SummonMinionCopy(target, "Mirror Image");
            SummonMinionCopy(target, "Mirror Image");
        }
    }
}